pyomo/gdp/disjunct.py

```diff
@@ -530,10 +515,12 @@ def set_value(self, expr):
             disjunct.constraint = c = ConstraintList()
             disjunct.propositions = p = LogicalConstraintList()
             for e in expressions:
-                if isinstance(e, BooleanValue):
+                if e.is_expression_type(ExpressionType.RELATIONAL):
+                    c.add(e)
+                elif e.is_expression_type(ExpressionType.LOGICAL):
                     p.add(e)
                 else:
-                    c.add(e)
+                    raise RuntimeError("Unsupported expression type on ")
```

Comment:
Where is the rest of this error message?

Reply:
Fixed.